sound/soc/sof/topology.c

```diff
@@ -1654,16 +1654,21 @@ static int sof_link_load(struct snd_soc_component *scomp, int index, struct snd_
 		return -ENOMEM;
 	}
 
-	/* parse one set of DAI link tokens */
-	ret = sof_copy_tuples(sdev, private->array, le32_to_cpu(private->size),
-			      SOF_DAI_LINK_TOKENS, 1, slink->tuples,
-			      num_tuples, &slink->num_tuples);
-	if (ret < 0) {
-		dev_err(scomp->dev, "failed to parse %s for dai link %s\n",
-			token_list[SOF_DAI_LINK_TOKENS].name, link->name);
-		goto err;
+	if (token_list[SOF_DAI_LINK_TOKENS].tokens) {
+		/* parse one set of DAI link tokens */
+		ret = sof_copy_tuples(sdev, private->array, le32_to_cpu(private->size),
+				      SOF_DAI_LINK_TOKENS, 1, slink->tuples,
+				      num_tuples, &slink->num_tuples);
+		if (ret < 0) {
+			dev_err(scomp->dev, "failed to parse %s for dai link %s\n",
+				token_list[SOF_DAI_LINK_TOKENS].name, link->name);
+			goto err;
+		}
 	}
 
+	if (!token_list[token_id].tokens)
+		goto out;
+
```

Comment:
commit message is very unclear
a) some parts of the dai link config cannot be read from NHLT, e.g. bclk and mclk rate
b) the NHLT may not even be present.

Reply:
i've removed the reference to NHLT. Basically if there's no need fo rDAI tokens, we shouldnt parse them like in the HDA case

Reply:
Regarding commit message and title, this is not just for IPC4, right? Any IPC type may not have DAI link tokens.

Reply:
Right, the HDA is only mentioned as an example